model/parts/pos_incentives.py

```diff
@@ -30,10 +30,11 @@ def policy_attestation_rewards(
     TIMELY_HEAD_WEIGHT = params["TIMELY_HEAD_WEIGHT"]
     WEIGHT_DENOMINATOR = params["WEIGHT_DENOMINATOR"]
 
+
     # State Variables
     base_reward = previous_state["base_reward"]
-    number_of_validators = previous_state["number_of_validators"]
-    number_of_validators_online = previous_state["number_of_validators_online"]
+    number_of_validators = previous_state["number_of_awake_validators"]
```

Comment:
@marthendalnunes For code review purposes:

I made the decision to keep `number_of_validators` variable naming to avoid making too many changes to code, and allow us to make variable name more generic to handle the validator cap being disabled.

I suggest we could refactor to the following to make more clear:
```python
if max_validator_count: (specify in parameter docstring that to disable max_validator_count feature you set to None)
  number_of_validators = previous_state["number_of_awake_validators"]
else:
  number_of_validators = previous_state["number_of_active_validators"]
```

Setting the max_validator_count to infinity has the same practical effect as disabling it. NB: We need to properly handle the parameter value being None or infinity where it's used in calculations.

Reply:
@BenSchZA this makes sense to me. We would need this logic in several places, right? shouldn't we make a function for it?

Reply:
@marthendalnunes that's a good idea - perhaps `get_number_of_validators(state) -> Int` in `model.parts.utils` using `__init__.py`?

Reply:
@BenSchZA it seems a good place, but the logic issue I see is that there are cases where we want the `number_of_awake_validators` and others where we want `number_of_active_validators ` (considering a max validator cap), so this function as we stated would be useful only when we want the `number_of_awake_validators`

Reply:
@marthendalnunes probably best handling logic on case by case basis then, if no general widely used function make sense.